### PR TITLE
[4/n][dagster-sigma] Build out translator class

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Introduces standard translator class to build Sigma specs. Pulls various pieces of metadata off assets, builds dependencies, and partial column schema for datasets.

## Test Plan

Unit tests.

## Changelog

> NOCHANGELOG

### CODE REVIEW

**Comment by benpankow** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24478?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#24478** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24478?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#24477** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24477?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24476** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24476?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24475** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24475?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @benpankow and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/libraries/dagster-sigma/dagster_sigma/translator.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            ),
+        }
+
+        return AssetSpec(
+            key=self.get_dataset_key(data),
+            metadata=metadata,
+            tags={
+                **build_kind_tag("sigma"),
+            },
+            deps={AssetKey(input_name.lower().split(".")) for input_name in data.inputs},
```
Comment (maximearmstrong) [inline python_modules/libraries/dagster-sigma/dagster_sigma/translator.py:113]: Should we create a `get_input_key()` method? Is an input a raw data source here?

**Review by maximearmstrong — APPROVED** [review]
One question, but it's not a blocker. Aside from that, lgtm!

**Comment by benpankow** [comment]
### Merge activity

* **Sep 19, 8:52 AM PDT**: @benpankow started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/24478).

**Pull request merged**